components/sceneutil/mwshadowtechnique.hpp

```diff
@@ -261,18 +261,18 @@ namespace SceneUtil {
 
         float                                   _shadowFadeStart = 0.0;
 
-        class DebugHUD : public osg::Referenced
+        class DebugHUD final : public osg::Referenced
         {
         public:
             DebugHUD(int numberOfShadowMapsPerLight);
 
-            virtual void draw(osg::ref_ptr<osg::Texture2D> texture, unsigned int shadowMapNumber, const osg::Matrixd &matrix, osgUtil::CullVisitor& cv);
+            void draw(osg::ref_ptr<osg::Texture2D> texture, unsigned int shadowMapNumber, const osg::Matrixd &matrix, osgUtil::CullVisitor& cv);
 
-            virtual void releaseGLObjects(osg::State* state = 0) const;
+            void releaseGLObjects(osg::State* state = 0) const;
```

Comment:
This is maybe an override. If it isn't, I maybe should have made the class extend `osg::Object` instead.

Reply:
Deriving from `osg::Object` requires to add more stuff here that is actually  not required. Only one object of this class is created in openmw code and it is used just to call defined methods. I'm not even sure it should be derived from `osg::Referenced`. `std::unique_ptr` should be enough because it's not shared anywhere.

Reply:
I didn't use `std::unique_ptr` because I've kept to OSG's code style in files taken from their repo.


While we're on the subject, IIRC there are some naked pointers nearby that should really be observer pointers, but aren't because nothing showed up when I googled for OSG weak pointer.

Reply:
Have the naked pointers been found and converted to observer pointers yet?

Reply:
IIRC, `osg::ref_ptr` and `std::unique_ptr` conflict with each other, so we use the former for OSG objects.